core/google/cloud/iterator.py

```diff
@@ -202,6 +202,14 @@ class Iterator(object):
                        takes the :class:`Iterator` that started the page,
                        the :class:`Page` that was started and the dictionary
                        containing the page response.
+
+    :type page_iter: callable
+    :param page_iter: (Optional) Callable to produce a pages iterator from the
+                      current iterator. Assumed signature takes the
+                      :class:`Iterator` that started the page. By default uses
+                      the HTTP pages iterator. Meant to provide a custom
+                      way to create pages (potentially with a custom
+                      transport such as gRPC).
```

Comment:
@tseaver This doesn't change the API surface (assuming end-users never touch the constructor), so not sure how you mean it is a big step backward. I am fine using a different class though.